### ISSUE TITLE
Move Many2one references from former module to new core

### ISSUE THREAD

**Issue by rousseldenis** [OCA/openupgradelib#309]
Case:

- A model (A) is defined in a custom module
- Some models have Many2one references to that model A.

- I want to move records of model A to new model in core Odoo (some INSERTs)
- Move the Many2one references to point to the new inserted records (different ids).

I could not find a helper here to do such things. Do you have an idea ?

@StefanRijnhart @hbrunn @pedrobaeza

**Comment by rousseldenis** [comment]
I've found this but is it sufficent ? 
https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade_merge_records.py

**Comment by pedrobaeza** [comment]
There's no such helper, as this is not a usual case AFAIK, and may be handled different depending on the feature. It's weird that you can simply replace one by the other. I don't have any case in mind like this in OpenUpgrade core addons.

**Comment by rousseldenis** [comment]
> There's no such helper, as this is not a usual case AFAIK, and may be handled different depending on the feature. It's weird that you can simply replace one by the other. I don't have any case in mind like this in OpenUpgrade core addons.

That's because Odoo is improving based on OCA work :smile:

**Comment by pedrobaeza** [comment]
Hehe. But we do a model and table renaming (and possibly field renaming) instead of insert queries, so no different IDs.

**Comment by rousseldenis** [comment]
> Hehe. But we do a model and table renaming (and possibly field renaming) instead of insert queries, so no different IDs.

That's not so trivial as records may exists in base module

**Comment by pedrobaeza** [comment]
Yeah, I remember such case with issues converted to tasks in v11 and we handle them manually. I think it's difficult to get to something reusable in a function, but if you want to try....

**Comment by rousseldenis** [comment]
> Yeah, I remember such case with issues converted to tasks in v11 and we handle them manually. I think it's difficult to get to something reusable in a function, but if you want to try....

I will for sure as soon I get complete solution
